src/synthesiser/Synthesiser.h

```diff
@@ -95,6 +101,19 @@ class Synthesiser {
         return it->second;
     }
 
+    /** Lookup symbol index */
+    size_t lookupSymbolIdx(const std::string& symbol) {
+        auto it = symbolMap.find(symbol);
+        if (it != symbolMap.end()) {
+            return it->second;
+        } else {
+            symbolIndex.push_back(symbol);
```

Comment:
`symbolIndex` is private and not used for anything other than this AFAICS.  Is it needed at all?  All the keys are captured as the keys of `symbolMap`, too.

Reply:
We need to print the symbols in order (symbol with number zero first and so forth.) for pre-loading the symbol table.  The code for the emitting the pre-load is here:

```
// issue symbol table with string constants
    visitDepthFirst(prog, [&](const StringConstant& sc) { convertSymbol2Idx(sc.getConstant()); });
    os << "SymbolTable symTable";
    if (!symbolMap.empty()) {
        os << "{\n";
        for (const auto& x : symbolIndex) {
            os << "\tR\"_(" << x << ")_\",\n";
        }
        os << "}";
    }
    os << ";";
```

Alternatively, we could avoid the vector and start querying for each symbol for a given index number. 
However, for large tables this might be very slow, i.e., we need a fast reverse lookup.

Reply:
Thanks for the clarification, that makes sense.  I must have misread/misunderstood the code when I was reviewing.